### ISSUE TITLE
The datasets module needs a better display

### ISSUE THREAD

**Issue by andresgsaravia** [andresgsaravia/research-engine#102]
For example:
- [x] The pages `/username/projectname/datasets` should list the available datasets in a friendlier way. Maybe we should be able to toggle the information in the same fashion as we do with each data concept. 
- [x] The dataset information needs to be parsed with gfm_markdown and mathjax.
- [x] Markdown and mathjax in data concepts' description.
- [x] Dataconcepts should be listed in the reverse way.
- [x] While uploading new datafiles, the _Upload new data_ button label is missleading.
